merge_duplicate_contacts/wizard/merge_partner_manual.py

```diff
@@ -557,9 +557,9 @@ def action_merge(self, context=None):
                 if this.vat_2:
                     self._cr.execute(
                         """
-                        UPDATE res_partner SET vat IN %s WHERE id IN %s
+                        UPDATE res_partner SET vat = %s WHERE id = %s
                         """,
-                        (tuple(this.vat_2), tuple(this.dst_partner_id.id)),
+                        (this.vat_2, this.dst_partner_id.id),
```

Comment:
The entire code would be way easier to understand if a naming convention for the fields would be used. e.g. `name_left`, `name_right` ... in that way everything would be iterable and you could use `getattr` and string manipulation to handle all cases. Less redundancy and more readable code. 95% of the form can also be automatically generated using `fields_view_get`.

At this point I'm confused why the UPDATE-query is necessary when you already wrote the vat in the previous statement.

I'm generally concered about the entire file. The entire file looks like redundant code when I look into the `base.partner.merge.wizard` but drops the update on foreign keys etc which could lead to missing references after you merge partner with this method. If you want to swap fields between similar partners it's ok but if you merge I would recommend the usage of the base function!

Reply:
@fkantelberg It might be relevant to note that I didn't write this file and am also confused by a lot of the decisions made in it, this is just me trying to fix the issues by modifying as little of it as possible. Maybe refactoring it entirely would be a good idea for a separate issue. (Also commented this on gitlab)